### ISSUE TITLE
I'd like a minimal configuration for Harbor

### ISSUE THREAD

**Issue by tengqm** [goharbor/harbor#17133]
**Is your feature request related to a problem? Please describe.**

I'm starting to feel frustrated at the footprint and the complexity of the Harbor installation.

**Describe the solution you'd like**

The default installation is very big today, thanks to the ever expanding scope of the project. We have JobService, Nginx, Portal, Triviy, ChartMuseum to install. We need external SQL database and Redis service...

Can we have a recommended minimal install configuration? I'm more interested in the container image service itself, nothing else.

**Describe the main design/architecture of your solution**

I don't have an architecture of the solution yet. The current design looks like a huge system for our environment.

**Describe the development plan you've considered**

If we are more focused on one thing and make it the best on earth, the plan can be very simple.

**Additional context**

I'm not expecting "more" components to be added. I'm expecting far less components/dependencies for managing container images.

**Comment by chlins** [comment]
@tengqm For helm installation the must required components are `core`, `jobservice`, `registry` and `portal`(if you do not need harbor UI, this is optional),  you could disable other unneeded components when you install.

**Comment by tengqm** [comment]
Thanks for the responses, folks. I'm not sure if I need `jobservice` at all. It smells like a component for sync'ing images across registries, however, it is now coupled with the core API layer (and maybe others). Is it possible to decouple it so that `jobservice` becomes optional?

**Comment by chlins** [comment]
Yes, the jobservice is for async job, and currently the jobservice is not the optional, but you can try to remove related deployment/svc/... manually if you don't need these async jobs scenarios?

**Comment by tculp** [comment]
I have an edge use case for a registry that only has the proxy cache functionality. No replication, image pushing, UI, etc. @qnetter how are those early discussions going?

**Comment by qnetter** [comment]
We have so far been looking at full copies at the edge, but proxy cache will likely be next pu in consideration.

Sent from my iPhone

On Jan 2, 2023, at 10:23 AM, tculp ***@***.***> wrote:

﻿
!! External Email

I have an edge use case for a registry that only has the proxy cache functionality. No replication, image pushing, UI, etc. @qnetter<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fqnetter&data=05%7C01%7Crklorese%40vmware.com%7C287228da0a3f4cb6ad7b08daecee6dbf%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C638082806043462366%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9UryYkfn7dceeSH%2FV7fdP%2BPHcQjQpZxrw0F4q5%2Bbn6E%3D&reserved=0> how are those early discussions going?

—
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgoharbor%2Fharbor%2Fissues%2F17133%23issuecomment-1369135703&data=05%7C01%7Crklorese%40vmware.com%7C287228da0a3f4cb6ad7b08daecee6dbf%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C638082806043462366%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Im1kqsY6%2B5bjpsSNzpL3Xod0gkTzsn%2FM9SJr5j3hlnI%3D&reserved=0>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACZWVL6OX6A4TXXOYVHW7T3WQMMJTANCNFSM52Y3K2KA&data=05%7C01%7Crklorese%40vmware.com%7C287228da0a3f4cb6ad7b08daecee6dbf%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C638082806043462366%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=3e8kMBLjgPLS09I7UwfLeYcYdGKtXvAesMpiS%2FSQk24%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

!! External Email: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender.

**Comment by aceeric** [comment]
I'd like to add something. One common need is to be able to sync from an upstream that requires two-way TLS with a client cert/key presented by Harbor and validated on the server, and of course a server cert/key presented by the server validated by a CA on the Harbor side. My best understanding is that this is not possible. If correct - is that also on the roadmap?
